### PR TITLE
CLOUDP-91945: Add support for endpoints v1.5

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/mongodbatlas.go [hunk 1]
```diff
@@ -64,6 +66,7 @@ type RequestDoer interface {
 	Doer
 	Completer
 	NewRequest(context.Context, string, string, interface{}) (*http.Request, error)
+	NewRequestAndSetBaseUrl(context.Context, string, string, string, interface{}) (*http.Request, error)
```
Comment (andreaangiolillo) [inline mongodbatlas/mongodbatlas.go:69]: The idea is to remove `NewRequest` and use only `NewRequestAndSetBaseUrl` that takes the API version as input

File: mongodbatlas/mongodbatlas.go [hunk 2]
```diff
@@ -80,6 +83,18 @@ type PlainRequestDoer interface {
...
+//	client    *http.Client
+//	BaseURL   *url.URL
+//	UserAgent string
+//
+//	// Services used for communicating with the API
+//	AdvancedClusters AdvancedClustersService
+//
+//	onRequestCompleted RequestCompletionCallback
+//}
+
```
Comment (andreaangiolillo) [inline mongodbatlas/mongodbatlas.go:97]: This was just an early attempt

File: mongodbatlas/mongodbatlas.go [hunk 3]
```diff
@@ -393,6 +428,15 @@ func (c *Client) NewGZipRequest(ctx context.Context, method, urlStr string) (*ht
...
 }
 
+// NewRequestAndSetBaseUrl creates an API request for atlas API.
+// A relative URL can be provided in urlStr, which will be resolved to the
+// BaseURL of the Client. Relative URLS should always be specified without a preceding slash.
+func (c *Client) NewRequestAndSetBaseUrl(ctx context.Context, apiVersion, method, urlStr string, body interface{}) (*http.Request, error) {
+	baseURL, _ := url.Parse(fmt.Sprintf(defaultBaseURL, apiVersion))
+	c.BaseURL = baseURL
+	return c.NewRequest(ctx, method, urlStr, body)
+}
```
Comment (andreaangiolillo) [inline mongodbatlas/mongodbatlas.go:438]: here I am setting the base URL

File: mongodbatlas/advanced_clusters.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+	if groupID == "" {
+		return nil, nil, NewArgError("groupId", "must be set")
+	}
+	if createRequest == nil {
+		return nil, nil, NewArgError("createRequest", "cannot be nil")
+	}
+
+	path := fmt.Sprintf(clustersPath, groupID)
+
+	req, err := s.Client.NewRequestAndSetBaseUrl(ctx, ApiV15, http.MethodPost, path, createRequest)
```
Comment (andreaangiolillo) [inline mongodbatlas/advanced_clusters.go:163]: This is an example of how to use it

File: mongodbatlas/mongodbatlas.go [hunk 5]
```diff
@@ -233,7 +249,7 @@ func NewClient(httpClient *http.Client) *Client {
 		httpClient = http.DefaultClient
 	}
 
-	baseURL, _ := url.Parse(defaultBaseURL)
+	//baseURL, _ := url.Parse(defaultBaseURL)
```
Comment (andreaangiolillo) [inline mongodbatlas/mongodbatlas.go:252]: we should’ve need this anymore

File: mongodbatlas/mongodbatlas.go (hunk 1 above)
Comment (gssbzn) [reply mongodbatlas/mongodbatlas.go:69]: oh I think I know what you want but no, let's not go this, this is now the time for something I've wanted a long time ago, let's make `defaultBaseURL="http://cloud.mongodb.com/"` and let each group of APIs include the `api/atlas/v1.0/` on their paths

Comment (andreaangiolillo) [reply mongodbatlas/mongodbatlas.go:69]: Got it! Thanks! : )

**Pull request closed without merging**